solution_template/jenkins/MainTemplate.json

```diff
@@ -0,0 +1,331 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "_artifactsLocation": {
+      "metadata": {
+        "description": "Artifacts location"
+      },
+      "type": "string"
+    },
+    "_artifactsLocationSasToken": {
+      "metadata": {
+        "description": "Artifacts Sas token"
+      },
+      "type": "string"
+    },
+    "adminPassword": {
+      "metadata": {
+        "description": "Password for the Virtual Machine. Will be used only is authenticationType is 'password'"
+      },
+      "type": "securestring"
+    },
+    "adminSSHPublicKey": {
+      "metadata": {
+        "description": "Public SSH key for the Virtual Machine. Will be used only is authenticationType is 'sshPublicKey'"
+      },
+      "type": "securestring"
+    },
+    "adminUserName": {
+      "metadata": {
+        "description": "User name for the Virtual Machine"
+      },
+      "type": "string"
+    },
+    "authenticationType": {
+      "metadata": {
+        "description": "Authentication type (can be 'password' or 'sshPublicKey')"
+      },
+      "type": "string"
+    },
+    "dnsPrefix": {
+      "metadata": {
+        "description": "Unique DNS Name for the Public IP used to access the Jenkins Virtual Machine."
+      },
+      "type": "string"
+    },
+    "publicIPName": {
+      "metadata": {
+        "description": "Public IP name"
+      },
+      "type": "string"
+    },
+    "publicIPNewOrExisting": {
+      "metadata": {
+        "description": "Should be 'new' if the public IP has to be created"
+      },
+      "type": "string"
+    },
+    "storageAccountName": {
+      "metadata": {
+        "description": "Storage Account Name"
+      },
+      "type": "string"
+    },
+    "storageAccountNewOrExisting": {
+      "metadata": {
+        "description": "Should be 'new' or 'existing'"
+      },
+      "type": "string"
+    },
+    "storageAccountType": {
+      "allowedValues": [
+        "Standard_LRS",
+        "Standard_GRS",
+        "Standard_RAGRS",
+        "Premium_LRS"
+      ],
+      "defaultValue": "Standard_LRS",
+      "metadata": {
+        "description": "Storage Account Type"
+      },
+      "type": "string"
+    },
+    "vmName": {
+      "metadata": {
+        "description": "Virtual Machine Name (also used as a prefix for other resources)"
+      },
+      "type": "string"
+    },
+    "vmSize": {
+      "metadata": {
+        "description": "Virtual Machine Size"
+      },
+      "type": "string"
+    }
+  },
+  "variables": {
+    "_extensionScript": "install_jenkins.sh",
+    "baseUrl": "[concat(parameters('_artifactsLocation'),'/solution_template/jenkins')]",
```

Comment:
I like to keep them here because it's easier to update them. If I'd inline them, then it would be harder to update them if required.
I do agree I should inline baseURL, but for the others I'd like to keep them like this